whylog/teacher/rule_validation_problems.py

```diff
@@ -0,0 +1,89 @@
+import itertools
+from collections import namedtuple
+
+ValidationResult = namedtuple('ValidationResult', ['errors', 'warnings'])
+
+
+class ValidationResult(object):
+    def __init__(self, errors, warnings):
+        """
+        :param errors: problems preventing form rule saving
+        :param warnings: problems accepted while rule saving
+        :type errors: [RuleValidationProblem]
+        :type warnings: [RuleValidationProblem]
+        """
+        self.errors = errors
+        self.warnings = warnings
+
+    @classmethod
+    def result_from_results(cls, results):
+        errors = sum([result.errors for result in results], [])
+        warnings = sum([result.warnings for result in results], [])
+        return ValidationResult(errors, warnings)
+
+    def acceptable(self):
+        return not self.errors
+
+    def select_parser_problems(self, line_id):
+        return [
+            problem for problem in itertools.chain(self.warnings, self.errors)
+            if isinstance(problem, ParserValidationProblem) \
```

Comment:
You should know what he needs this for. If it is to determine the order, then the objects should be made comparable so that he could sort them and display them in a sorted order.

I guess he needs to assign parser problems to their id's to display the warnings. In such case Parser and Constraint should be made hashable and your layer should return a dictionary `<problematic_object, list_of_problems>`, where a problematic_object can be a Parser or a Constraint.
That would be much better than your layer splitting the objects in two groups, so that the front can loop over the list and reconstruct the mappings...